### DISCUSSION TITLE
Question about the finetuning and model configuration

### DISCUSSION THREAD

**Discussion by jsw-zorro** [tensorflow/compression#85] [Q&A]
Hi, I am wondering is there any convenient way to finetune the pre-trained model on a new dataset? Or is it convenient for you to provide the model parameter setup for the different pre-trained models? I am wondering about different parameter setup for the models ranging from `bmshj2018-hyperprior-msssim-1` to `bmshj2018-hyperprior-msssim-8`.

Thanks for help in advance.

**Comment by jonaballe** [comment]
Hi, unfortunately our published pre-trained models can't easily be fine tuned. To make the computations cross-platform compatible and prevent them from being dependent on specific TensorFlow versions, we provide them as computation graphs that have been constant-folded, with all unnecessary parts removed. So, no TensorFlow variables remain in the graph, and it doesn't directly map to Python code anymore. Also, the variables representing the continuous versions of the probability models are missing from the graphs, since they are not needed during inference time. So even if you reverse engineered the graphs to extract the kernels/biases, there would be no way to continue training the differential proxy for the rate. You'd have to reinitialize that part of the model.

Right now, I think your best option would be to train a model from scratch, based on the provided code in the `models/` directory.

**Reply by jsw-zorro** [reply]
Get it, thx!
